### PR TITLE
feat(email): email attachment claim optimization

### PR DESCRIPTION
**Pull request by whutchinson98**
# Changes
1. Centralized attachment filtering
  - Added shared document MIME types, octet-stream extensions, and media MIME prefixes.
  - Generated existing SQL filters from those allowlists.
  - Added Rust helpers for matching document and media attachments.
  - Added parity and edge-case tests.
2. Gated inbox-sync claim calls
  - Inbox sync now classifies attachments before accessing the database.
  - Skips all claim queries when no eligible attachments exist.
  - Independently skips document or media queries when that category is absent.
  - Preserved feature flags, concurrent query structure, event metadata, and upload behavior.
  - Added classification tests for document, media, mixed, unsupported, and incomplete attachments.
3. Added a document candidate probe
  - Before running the expensive condition-5 query, checks whether the message has an eligible, unclaimed, not-yet-uploaded document attachment.
  - Returns early when no candidate exists.
  - Keeps the actual claim query race-safe and unchanged.
  - Added database tests for claimed, uploaded, unsupported, missing-filename, unknown-message, and mismatched-link cases.

---

Reduce unnecessary and expensive database work during email inbox sync, especially attachment claim queries, without changing which valid attachments are ultimately uploaded.

## Expected results
- Signature assets, unsupported inline parts, and emails without eligible attachments trigger no attachment claim queries.
- Document-only emails skip media queries; media-only emails skip document queries.
- Already claimed, already uploaded, or otherwise ineligible documents avoid the expensive condition-5 contacted-set query.
- PDF, DOCX, image, and video attachments continue to be claimed and uploaded normally.
- Repeated inbox-sync processing becomes significantly cheaper.

Operationally, this should produce:
 - Lower condition-5 query execution volume.
 - Reduced PostgreSQL buffer I/O and database load.
 - Faster inbox-sync processing and less contention.
 - Improved scalability during inbox-sync-heavy periods.
 - No user-visible attachment upload regressions.

### CODE REVIEW

File: crates/email_db_client/src/attachments/provider/upload.rs [hunk 1]
```diff
@@ -220,6 +220,40 @@ pub async fn fetch_job_attachments_for_backfill(
     Ok(attachments)
 }
 
+async fn message_has_unclaimed_document_attachment(
+    db: &Pool<Postgres>,
+    link_id: Uuid,
+    message_provider_id: &str,
+) -> anyhow::Result<bool> {
```
[the remark this answers is not in the document]
Comment (whutchinson98) [reply crates/email_db_client/src/attachments/provider/upload.rs:227]: this crate predates rootcause

**Comment by evanhutnik** [comment]
Nice — the parity test on the generated SQL makes layer 1 easy to trust, and gating on the parsed part list before touching the DB is the right place to catch the signature-image case.

One suggestion on the probe placement:

`message_has_unclaimed_document_attachment` is called at `upload.rs:101`, below the `!attachments.is_empty()` early return, so it only guards query 2. But its predicates (`m.link_id`, `m.provider_id`, `de.email_attachment_id IS NULL`, `upload_claimed_at IS NULL`, `filename IS NOT NULL`, document MIME filter) are exactly query 1's claim-set predicates minus the thread-wide `EXISTS` — so `has_candidate = false` implies query 1 claims nothing either.

Hoisting the probe to the top of `new_email_document_atts`, before query 1, looks safe and covers the case the PR description calls out as the main win. Right now, re-processing a message whose document attachments are already claimed or already have a `document_email` row still pays for query 1's `EXISTS` over the whole thread with the `email_message_labels` / `email_labels` / `email_contacts` joins before reaching the probe. Since inbox-sync redelivery is the repeated case, that's probably where a good chunk of the remaining buffer I/O lives.

Smaller note: `attachment_upload_eligibility` classifies from the parsed Gmail resource's `mime_type` / `filename`, while the claim queries read `email_attachments`. Same data today, but if those ever drift, the failure mode is a silently skipped upload rather than a redundant one. Might be worth a one-liner at the call site noting the gate must stay a superset of the SQL filters, so a future change to one side doesn't quietly narrow the other.
